scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -344,6 +350,32 @@ public void initActivity(Body body) {
         }
     }
 
+    private void releaseBusyNodesWithNoRunningTask(RMProxy rmProxy, RecoveredSchedulerState recoveredState) {
+
+        Vector<InternalJob> runningJobs = recoveredState.getRunningJobs();
+
+        List<NodeSet> busyNodesWithTask = findBusyNodesCorrespondingToRunningTasks(runningJobs);
+
+        rmProxy.releaseDanglingBusyNodes(busyNodesWithTask);
+    }
+
+    private List<NodeSet> findBusyNodesCorrespondingToRunningTasks(Vector<InternalJob> runningJobs) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)